UM/Trust.py

```diff
@@ -384,6 +384,81 @@ def _verifyFile(self, filename: str, signature: str) -> bool:
+            if self_sign is None:
+                return False
+            shash = TrustBasics.getSelfSignHash(signatures_json)
+            if shash is None or not self._verifyHash(shash, self_sign):
+                return False
+            return True
+        except:  # Yes, we do really want this on _every_ exception that might occur.
+            return False
+
+    def signedFolderPreStorageCheck(self, path: str) -> bool:
+        """Do a quick check whether the 'central storage file' of a folder has been tampered with. This is necessary,
+        since the central storage system (which otherwise runs first) copies files, and the copying of files itself
+        can be an attack. Note that right after copying, a full check can be done, so the files themselves don't have
+        to be checked yet (since that happens in the full check after copying).
+
+        Shared pools of versioned items ('central storage') are used if a folder contains a 'central storage file'.
+        (See the CentralFileStorage class for details.) The 'canonical' version of a folder as far as the Trust system
+        is concerned, is the one where the items are already in central storage. Otherwise there would either be a whole
+        range of 'acceptable answers' (and that's harder to test against) or, there would always be a need to verify
+        a situation that shouldn't be that frequent (items need to be copied to central storage). This creates a problem
+        in that the central storage mechanism needs to run _first_, however. The central storage has it's own security
+        measures, but this means that the central storage file in a folder (which contains info on what items should be
+        copied) hasn't been checked against the trust manifest file in that folder yet.
+
+        This only concerns the signature of the central storage file (and the correctness of the manifest file itself).
+        Per separation of concern, and since the storage system already needs to be aware of security, any other
+        'sanity checks' on the contents central storage file itself are the job of that system.
+
+        :param path: The folder to do a quick pre-move check for.
+        :return: True if the central-storage file is correctly signed. A folder without such a file is correct as well.
+        """
+
+        try:
+            # Check if the central storage file exist, if not, then the system won't copy anything in any case.
+            central_storage_filename = os.path.join(path, TrustBasics.getCentralStorageFilename())
+            if not os.path.exists(central_storage_filename):
+                Logger.log("i", f"No central storage file for unbundled folder '{path}'.")
+                return True
+
+            # Open the file containing signatures (just reading the json is negligible compared to the verify or store):
+            json_filename = os.path.join(path, TrustBasics.getSignaturesLocalFilename())
```

Comment:
Could we give these a bit more descriptive variable names? It's quite a blur right now. Something like:

`json_filename` -> `signatures_filename` or `signatures_path`
`data_file` -> `signatures_file`
`json_data` -> `signatures_data` or `signatures_contents`
`signatures_json` -> `root_signature_category` (it's not JSON any more at this point, just a `dict`).
`name_in_data` -> `central_storage_filename`

Feel free to come up with different names of course. But the point is that the variables describe what they contain, not just the data type. That would make this segment a lot easier to understand.

Reply:
Resolved in 6f7dadba